pennylane/tape/qscript.py

```diff
@@ -434,10 +438,11 @@ def _update(self):
         self._update_par_info()  # Updates _par_info; O(ops+obs)
 
         self._update_observables()  # Updates _obs_sharing_wires and _obs_sharing_wires_id
-        self._update_batch_size()  # Updates _batch_size; O(ops)
-
-        # The following line requires _batch_size to be up to date
-        self._update_output_dim()  # Updates _output_dim; O(obs)
+        if self._batch_size is not _UNSET_BATCH_SIZE:
+            # this avoids computing it on init (before it has been requested, lazy)
+            self._update_batch_size()  # Updates _batch_size; O(ops)
+            # The following line requires _batch_size to be up to date
+            self._update_output_dim()  # Updates _output_dim; O(obs)
```

Comment:
Do we need these lines at all anymore?

Reply:
I just figured that someone might call `_update` directly on a tape that's already evaluated the batch size, but I guess we should continue to lazy-load, even if they've done this strange order of events